.github/renovate.json5

```diff
@@ -0,0 +1,57 @@
+{
+    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
+    "extends": [
+        "config:recommended",
+        ":dependencyDashboardApproval",
+        ":semanticCommitScopeDisabled"
+    ],
+    "internalChecksFilter": "strict",
+    "labels": ["dependencies"],
+
+    // Wait a few days for any new package, as a precaution against malicious publishes
+    "minimumReleaseAge": "3 days",
+
+    "packageRules": [
+        // Enable major version updates, but don't auto-merge them
+        {
+            "automerge": false,
+            "matchUpdateTypes": ["minor"]
+        },
+
+        // Enable minor version updates with auto-merging
+        {
+            "automerge": true,
```

Comment:
How does automerge work with protected branches? Is it exempt?

Reply:
No, bots don't get any exemptions and need to follow the branch protections like any other user. So e.g. if a PR requires reviews, there needs to be a PR review. If you restrict who can merge to main branch and the bot isn't on the list, it won't be able to, etc.

Reply:
Okay, so for automerge to work we'd need to explicitly allow the bot permission to merge, otherwise the PR would remain open?

Reply:
yes

Reply:
Read the [Renovate docs, key concepts page for automerge](https://docs.renovatebot.com/key-concepts/automerge/) to learn more.

Reply:
Just confirming - is there any code change that needs to be done in this PR for this thread?